STM32G473CBTx/makefile

```diff
@@ -0,0 +1,212 @@
+######################################
+# target
+######################################
+TARGET = main
+
+######################################
+# building variables
+######################################
+# debug build?
+DEBUG = 1
+# optimization
+OPT = -Og
+
+
+#######################################
+# paths
+#######################################
+# Build path
+
+OBJECTS_DIR = $(BUILD_DIR)/objects
+PLATFORM_LIB_DIR = $(BUILD_DIR)/platform
+STANDALONE_DIR = $(BUILD_DIR)/standalone
+
+######################################
+# source
+######################################
+# C sources
+
+MAIN_SOURCE := ./Core/Src/main.cpp
+
+PLATFORM_SOURCES := $(shell find ../platform/src -name "*.c")
+
+CORE_C_SOURCES := $(shell find ./Core/Src -name "*.c")
+CORE_CPP_SOURCES := $(shell find ./Core/Src -name "*.cpp" | grep -v "main.cpp")
+
+HAL_SOURCES := $(shell find ./Drivers/STM32G4xx_HAL_Driver -name "*.c")
+
+RTOS_SOURCES = \
+./Middlewares/Third_Party/FreeRTOS/Source/croutine.c \
+./Middlewares/Third_Party/FreeRTOS/Source/event_groups.c \
+./Middlewares/Third_Party/FreeRTOS/Source/list.c \
+./Middlewares/Third_Party/FreeRTOS/Source/queue.c \
+./Middlewares/Third_Party/FreeRTOS/Source/stream_buffer.c \
+./Middlewares/Third_Party/FreeRTOS/Source/tasks.c \
+./Middlewares/Third_Party/FreeRTOS/Source/timers.c \
+./Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c \
+./Middlewares/Third_Party/FreeRTOS/Source/portable/MemMang/heap_4.c \
+./Middlewares/Third_Party/FreeRTOS/Source/portable/GCC/ARM_CM4F/port.c
+
+C_SOURCES = $(PLATFORM_SOURCES) $(CORE_C_SOURCES) $(HAL_SOURCES) $(RTOS_SOURCES)
+CPP_SOURCES = $(MAIN_SOURCE) $(CORE_CPP_SOURCES)
+
+# ASM sources
+ASM_SOURCES = ./Core/Startup/startup_stm32g473cbtx.s
+
+#######################################
+# binaries
+#######################################
+PREFIX = arm-none-eabi-
+# The gcc compiler bin path can be either defined in make command via GCC_PATH variable (> make GCC_PATH=xxx)
+# either it can be added to the PATH environment variable.
+ifdef GCC_PATH
+CC = $(GCC_PATH)/$(PREFIX)gcc
+AS = $(GCC_PATH)/$(PREFIX)gcc -x assembler-with-cpp
+CP = $(GCC_PATH)/$(PREFIX)objcopy
+SZ = $(GCC_PATH)/$(PREFIX)size
+AR = $(GCC_PATH)/$(PREFIX)ar
+CPP_CC = $(GCC_PATH)/$(PREFIX)g++ -std=c++11
+else
+CC = $(PREFIX)gcc
+AS = $(PREFIX)gcc -x assembler-with-cpp
+CP = $(PREFIX)objcopy
+SZ = $(PREFIX)size
+AR = $(PREFIX)ar
+CPP_CC = $(PREFIX)g++ -std=c++11
```

Comment:
Is there a reason that we are only tagetting c++11?

Reply:
Not really. I know that C++ 20 depreceated compound assignment to voltaile (which is something we need), so I just picked something older than that.
- https://www.reddit.com/r/cpp/comments/jswz3z/compound_assignment_to_volatile_must_be/

If we ever need anything later than C++ 11 we can always just change that variable tho.